packages/new-deploy/src/DeployContext.ts

```diff
@@ -113,29 +113,97 @@ export default class DeployContext extends MultiProvider<config.Domain> {
     this._data = config.addBridge(this.data, name, bridge);
   }
 
-  pushVerification(
-    nameOrDomain: string | number,
-    verification: Verification,
-  ): void {
-    const name = this.resolveDomainName(nameOrDomain);
-    if (!this.verification.has(name)) this.verification.set(name, []);
+  /// Adds a connection between two known networks. Use this AHEAD OF calling
+  /// `ensureConnections
+  addConnection(left: string, right: string): void {
+    // asserts that the domains are known
+    this.mustGetDomain(left);
+    this.mustGetDomain(right);
 
-    const net = this._verification.get(name);
-    if (net) net.push(verification);
+    if (this._data.protocol.networks[left].connections.indexOf(right) !== -1) {
+      throw new Error(`${right} already a connection of ${left}`);
+    }
+    if (this._data.protocol.networks[right].connections.indexOf(left) !== -1) {
+      throw new Error(`${left} already a connection of ${right}`);
+    }
+
+    this._data.protocol.networks[left].connections.push(right);
+    this._data.protocol.networks[left].connections.push(left);
   }
 
-  mustGetVerification(nameOrDomain: string | number): readonly Verification[] {
-    const domain = this.resolveDomainName(nameOrDomain);
-    const verification = this.verification.get(domain);
-    if (!verification)
-      throw new Error(
-        `Verification with name ${nameOrDomain} for domain ${domain} is not defined`,
-      );
+  // Intended entrypoint.
+  async deployAndRelinquish(): Promise<ethers.PopulatedTransaction[]> {
+    // validate the config input
+    this.validate();
 
-    return verification;
+    // ensure the presence of all core + bridge contracts, and enroll all core + bridge contracts with each other
+    const governanceTransactions = await this.ensureCoreAndBridgeConnections();
+
+    // relinquish control of all other contracts from deployer to governance
+    await this.relinquishOwnership();
+
+    // appoint governor on all networks
+    await Promise.all(
+      this.networks.map((network) =>
+        this.mustGetCore(network).appointGovernor(),
+      ),
+    );
+
+    // perform validation checks on core and bridges
+    await this.checkCores();
```

Comment:
TODO: this should not be called in the same process as the deploy script. in production, the governance txs need to go through before the checks can pass. and this fn should return the governance txs, rather than throwing